aten/src/ATen/native/Embedding.cpp

```diff
@@ -0,0 +1,181 @@
+#include "ATen/ATen.h"
+#include "ATen/Check.h"
+#include "ATen/NativeFunctions.h"
+
+#include <cstring>
+#include <memory>
+#include <sstream>
+#include <vector>
+
+#ifdef _OPENMP
+#include <omp.h>
+#endif
+
+
+namespace at { namespace native {
+
+Tensor embedding(const Tensor & weight, const Tensor & indices,
+                 int64_t padding_idx, bool scale_grad_by_freq, bool sparse) {
+  auto indices_arg = TensorArg(indices, "indices", 1);
+  checkScalarType("embedding", indices_arg, kLong);
+  checkContiguous("embedding", indices_arg);
```

Comment:
previously, we would set these to contiguous if they weren't; should we do that in the functional call for backwards compat?

Reply:
I'll add it to the Python code `def embedding`